### DISCUSSION TITLE
Firestore.FieldName marked as error

### DISCUSSION THREAD

**Discussion by st-trade** [AntlerVC/firetable#781] [Support / Q&A] [answered]
I my understanding, `db` in the extensions body, means the projects FirebaseFirestore instance. I'm not quite sure why FieldValue as in [FiresbaseFirestore.FieldValue](https://googleapis.dev/nodejs/firestore/latest/FieldValue.html) is marked as an error. 

![image](https://user-images.githubusercontent.com/94030557/187570486-b45f17cb-27c7-4320-8326-0dfb95a400f0.png)

**Comment by harinij** [comment]
Hi @st-trade 

`db` here gives you full access to all collections and documents in your entire Firestore project.

Here is an example of how you can use `db` in your derivative or extensions code. 

```
const derivative: Derivative = async ({ row, db }) => {
  // Get user doc
  const userDoc = db.collection("users").doc(row.uid);
  return userDoc.get("email");
};
```

In case you are looking for retrieve field of the current collection's row, then you can use `row.fieldname`. Hope that answers your questions.

**Comment by shamsmosowi** [comment, marked as answer]
not sure if FieldValue is exported in `db`, to access fieldValue object in extensions or derivatives you can require it
```javascript
  const { FieldValue } = require("firebase-admin/firestore");
  ```
while if you want you use it in webhooks which is esm only, you can import it like so
```javascript
 const {FieldValue} = await import('@google-cloud/firestore');
```
